src/App.tsx

```diff
@@ -1,34 +1,62 @@
 import { useKeycloak } from "@react-keycloak/web";
-import { Route, Routes, Navigate } from 'react-router-dom';
+import { Route, Routes, Navigate } from "react-router-dom";
 import { useEffect, useState } from "react";
-import { api } from './api/api';
-import Home from './pages/Home';
-import CaretakerForm from './pages/CaretakerForm';
-import LoginPage from './pages/LoginPage';
-import store from './store/RootStore';
+import { api } from "./api/api";
+import Home from "./pages/Home";
+import CaretakerForm from "./pages/CaretakerForm";
+import LoginPage from "./pages/LoginPage";
+import store from "./store/RootStore";
 import CaretakerSearch from "./pages/CaretakerSearch";
+import ProfileSelection from "./pages/ProfileSelection";
 
 function App() {
   const { keycloak, initialized } = useKeycloak();
+  const [isXsrfTokenFetched, setIsXsrfTokenFetched] = useState(false);
+  const [isUserDataFetched, setIsUserDataFetched] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
     const fetchXsrfToken = async () => {
       if (keycloak.authenticated && !store.user.xsrfToken) {
         await api.getXsrfToken();
+        setIsXsrfTokenFetched(true);
+      }
+    };
+
+    const fetchUserData = async () => {
+      try {
+        const userData = await keycloak.loadUserProfile();
+        const userProfileData = {
+          username: userData.username,
+          email: userData.email,
+          firstName: userData.firstName,
+          lastName: userData.lastName,
+          token: store.user.xsrfToken,
+          selected_profile: null,
```

Comment:
Nie możemy ustawiać `selected_profile` na `null`, bo przy każdym odświeżeniu strony będzie to ustawiało i prosiło użytkownika o wybranie profilu. Będzie trzeba pobrać w tym miejscu `user_profile` z tego co aktualnie jest w local storage i dopiero zapisać.

Reply:
Tak, ale przy odświeżeniu też się wykona to co jest w tym useEffect, bo wykonuje się zawsze przy ponownym zamontowanie komponentu, niezależnie co będzie w tablicy zależności. Z resztą możesz sam to przetestować i zobaczyć.

Reply:
O, ok. To sprawdzę

Reply:
Włożyłem funkcję fetchUserData w ifa. Co myślisz? [[put fetchUserData into if statement](https://github.com/Daarkosss/petBuddy-frontend/pull/11/commits/c0ecbdbf5efee0c63c488813a891e3bc675d34bc)](url)

Reply:
Myśle że okej